src/Model/Traits/HasOtp.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ecodev\Felix\Model\Traits;
+
+use Doctrine\ORM\Mapping as ORM;
+use Exception;
+use GraphQL\Doctrine\Annotation as API;
+use OTPHP;
+
+/**
+ * Trait for a user with second factor authentication using OTP.
+ */
+trait HasOtp
+{
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default" = 0})
+     */
+    private $otp = false;
+
+    /**
+     * @API\Exclude
+     *
+     * @ORM\Column(type="string", length=255, nullable=true)
+     */
+    private ?string $otpUri = null;
+
+    /**
+     * Must be implemented by the user class to use the login as the key label.
+     */
+    abstract public function getLogin(): ?string;
+
+    /**
+     * Enable 2FA for the user
+     * This should be only enabled after otpUri has been generated, stored and verified.
+     *
+     * @API\Exclude
+     */
+    public function setOtp(bool $otp): void
+    {
+        if ($otp === true && !empty($this->otpUri)) {
+            throw new Exception('Cannot enable OTP without a secret');
+        }
+
+        $this->otp = $otp;
+    }
+
+    /**
+     * Whether the user has 2FA enabled.
+     */
+    public function isOtp(): bool
+    {
+        return $this->otp;
+    }
+
+    /**
+     * Returns the OTP provisionning URI (to display QR code).
+     *
+     * @API\Exclude
+     */
+    public function getOtpUri(): ?string
+    {
+        return $this->otpUri;
+    }
+
+    /**
+     * Generate and store a new OTP secret.
+     *
+     * @param string $issuer identify the service that provided the OTP (application or host name)
+     * @param int $period number of seconds an OTP will be valid
+     * @param string $digest digest algorithm: sha1 or sha256, but Google Authenticator only works with sha1
+     * @param int $digits length of the OTP numeric code
+     *
+     * @API\Exclude
+     */
+    public function createOtpSecret(string $issuer, int $period = 30, string $digest = 'sha1', int $digits = 6): void
+    {
```

Comment:
J'aurais tendance à pas créer des paramètres qui n'ont pas d'utilité immédiate.

Un peu dans la philosophie de [PASSWORD_DEFAULT](https://www.php.net/manual/en/function.password-hash.php), il me préférable de garder ces options comme détails d'implémentations. Ainsi on conserve une liberté totale pour mettre à jour Felix dans le futur avec les nouvelles best-practice. Sans devoir se soucier d'éventuelle souci de compatibilité avec les utilisations existantes.

tl;dr: une surface d'API minimale est toujours plus facile à maintenir


```suggestion
    public function createOtpSecret(string $issuer): void
    {
        $period = 30;
        $digest = 'sha1;
        $digits = 6;
```